python/pdo/service_client/servicedatabase.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/env python
+
+# Copyright 2019 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# 
+
+import os
+import sys
+import json
+import copy
+
+import logging
+logger = logging.getLogger(__name__)
+
+from pdo.service_client.enclave import EnclaveServiceClient
+from pdo.service_client.storage import StorageServiceClient
+from pdo.common.utility import are_the_urls_same
+
+
+class ServiceDB_Manager():
+    """ A class to wrap calls to service database manager. 
+    
+    The data is saved as a json file, and is loaded as a dictionary of (key, value) pairs.  Key is a client chosen short name for the service.
+    Value is a dictionary with entries corresponding to id (if service is eservice), url, verifying_key (if service is sservice).  
+    
+    It is also possible to first create an empty dictonary, add entries and then save as a json file.
+    
+    Name, id (enclave_id) and url are all synonyms for a given eservice. Exceptions will be raised if there are violations. To fix a broken database, 
+    use the remove_service method to remove multiple or replicated entries for a given identifier from the database. After remove, use the add_new_entry
+    to add a unique entry for a given identifier"""
+
+    def __init__(self, file_name = None, service_type='eservice'):
+
```

Comment:
not sure that making this so general that it can handle multiple entry types is useful. in particular, you have no actual validation of the entries (other than "can i get the url"). make this an eservice db so that we can add the ability to pull the eservice info from the sawtooth ledger & validate it.

Reply:
i was only hoping to reusue the class for eservice and sservice. Ok, will make it unique to eservice, we can think about sservice later, given that the current PR does not anyway solve sserivce fully.